internal/node/hybrid/kubelet_cert_validator.go

```diff
@@ -13,51 +13,108 @@ import (
 	"github.com/aws/eks-hybrid/internal/kubelet"
 )
 
+// ValidationError represents a kubelet certificate validation error with remediation information
+type ValidationError struct {
+	Message          string
+	Remediation      string
+	Cause            error
+	IsDateValidation bool
+	NoCert           bool
+}
+
+func (e *ValidationError) Error() string {
+	if e.Cause != nil {
+		return fmt.Sprintf("%s: %v", e.Message, e.Cause)
+	}
+	return e.Message
+}
+
+func (e *ValidationError) Unwrap() error {
+	return e.Cause
+}
+
 // ValidateKubeletCert checks if there is an existing kubelet certificate and validates it against the cluster's CA
-func ValidateKubeletCert(logger *zap.Logger, installRoot string, ca []byte) error {
+func ValidateKubeletCert(logger *zap.Logger, installRoot string, ca []byte) *ValidationError {
```

Comment:
not a blocker
In go is a bit more common to just return the `error` interface and offer exported methods to check the "type" of error

That allows to keep a common/consistent method signature across methods, to not add complexity for most callers while still giving the option to check the type of errors to the callers who care

you could just have

```go

func hybrid.IsDateValidationError(err error) bool

func hybrid.IsNoCertError(err error) bool
```